RecoCTPPS/PixelLocal/interface/RPixDetPatternFinder.h

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+* This is a part of CTPPS offline software.
+* Author:
+*   Fabrizio Ferro (ferro@ge.infn.it)
+*   Enrico Robutti (robutti@ge.infn.it)
+*   Fabio Ravera   (fabio.ravera@cern.ch)
+*
+*/
+#ifndef RecoCTPPS_PixelLocal_RPixDetPatternFinder_H
+#define RecoCTPPS_PixelLocal_RPixDetPatternFinder_H
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/Common/interface/DetSet.h"
+
+#include "DataFormats/CTPPSReco/interface/CTPPSPixelRecHit.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h"
+
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSGeometry.h"
+#include "CLHEP/Vector/ThreeVector.h"
+#include "CLHEP/Vector/RotationInterfaces.h"
+#include "TMatrixD.h"
+
+#include <vector>
+
+class RPixDetPatternFinder{
+  
+public:
+  RPixDetPatternFinder(edm::ParameterSet const& parameterSet): parameterSet_(parameterSet) {}
+  
+  virtual ~RPixDetPatternFinder();
+
+  typedef struct{
+    CLHEP::Hep3Vector globalPoint;
+    TMatrixD          globalError;
+    CTPPSPixelRecHit  recHit     ;
+    uint32_t          detId      ; //I cannot use directy the CTPPSPixelDetId because it needs to be declared
```

Comment:
If CTPPSPixelDetId is useful, why not add it ?

Reply:
@slava77 
I used an uint32_t because CTPPSPixelDetId does not have a default constructor and the compiler was complaining in RPixRoadFinder.cc when I declare the thePointAndRecHit at line 54